src/common/utils/time-utils.test.ts

```diff
@@ -115,3 +117,50 @@ describe('subtract time', () => {
     );
   });
 });
+
+describe('get start of time', () => {
+  test('Start of hour', () => {
+    expect(getStartOf(new Date(2020, 2, 1, 0, 30), TimeUnit.HOURS)).toEqual(
+      testDate,
+    );
+  });
+  test('Start of day', () => {
+    expect(getStartOf(new Date(2020, 2, 1, 8, 30), TimeUnit.DAYS)).toEqual(
+      testDate,
+    );
```

Comment:
I think it's better to not use variables in test cases with custom input to make it easier for us to compare the expected input and output.

```ts
expect(getStartOf(new Date(2020, 2, 1, 8, 30), TimeUnit.DAYS))
  .toEqual(new Date(2020, 2, 1, 0, 0));
```

If a group of functions share the input value (the formatting functions, for example), it might make sense to use a variable for the input. Consider moving `testDate` to be inside `describe('time formatting', () => { ... })` block